Mathlib/LinearAlgebra/Eigenspace/Matrix.lean

```diff
@@ -0,0 +1,75 @@
+/-
+Copyright (c) 2024 Jon Bannon. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Jon Bannon, Jireh Loreaux
+-/
+
+import Mathlib.LinearAlgebra.Eigenspace.Basic
+
+/-!
+
+# Eigenvalues, Eigenvectors and Spectrum for Matrices
+
+This file collects results about eigenvectors, eigenvalues and spectrum specific to matrices
+over a field.
+
+## Tags
+eigenspace, eigenvector, eigenvalue, eigen, spectrum, matrices, matrix
+
+-/
+
+section SpectrumDiagonal
+
+variable {R : Type*} [Field R] {n : Type*} [DecidableEq n] [Fintype n] (M : Type*)
+  [AddCommGroup M] [Module R M]
+
+open Matrix
+open Module.End
+
+/-- Basis vectors are eigenvectors of associated diagonal linear operator. -/
+lemma hasEigenvector_toLin_basis_diagonal (d : n → R) (i : n) (b : Basis n R M) :
+    Module.End.HasEigenvector (toLin b b (diagonal d)) (d i) (b i) := by
+  constructor
+  · refine mem_eigenspace_iff.mpr ?left.a
+    simp only [diagonal, Pi.basisFun_apply, toLin'_apply, mulVec_stdBasis_apply, transpose_apply,
+      of_apply, Pi.smul_apply, LinearMap.stdBasis_apply', smul_eq_mul, mul_ite, mul_one, mul_zero]
+    all_goals simp_all
+  · exact Basis.ne_zero b i
+
+/--  Standard basis vectors are eigenvectors of any associated diagonal linear operator. -/
+lemma hasEigenvector_toLin'_diagonal (d : n → R) (i : n) :
+    Module.End.HasEigenvector (toLin' (diagonal d)) (d i) (Pi.basisFun R n i) :=
+  hasEigenvector_toLin_basis_diagonal (d : n → R) (i : n) (b := Pi.basisFun R n)
+
+/-- Eigenvalues of a diagonal linear operator are the diagonal entries. -/
+lemma hasEigenvalue_toLin'_diagonal_iff (d : n → R) {μ : R} :
```

Comment:
This should have a `toLin` generalization.

Reply:
Ah! I just learned something new. So, `toLin` is the general basis version, and `toLin'` is the `PiLp 2` version? 

Thanks for the clarification on this. (I assume this was what your previous suggestion about basis _ _ _ was referring to?)

Reply:
Actually, `toLin'` is just the standard basis version, but with the max norm. `toEuclideanLin` is the `PiLp 2` version.